### ISSUE TITLE
Add `root_id` to Document

### ISSUE THREAD

**Issue by JohannesMessner** [docarray/docarray#461]
`root_id` is like `parent_id`, but instead of pointing one level above, it points all the way to the top (root).

https://github.com/jina-ai/docarray/pull/460

**Comment by JohannesMessner** [comment]
We decided that the overhead is no worth the benefit in this case, since most scenarios do not require deep nesting.
However, there is one exception: list-types in multi-modal documents look like a single nesting to the user, but are actually two levels of nesting under the hood. This means that there is no straightforward way to get the root doc from a list doc.

So the new plan is the following:

- We make `root_id` as multimodal-only feature
- This will be a `@property` so we don't have to change the proto. Under the hood it will access a field in `_metadata`
- The information in `_metadata` is set only once, when the dataclass is converted to Document. We make no guarantees about any behaviour if the user changes the document tree structure afterwards

**Comment by JoanFM** [comment]
> We decided that the overhead is no worth the benefit in this case, since most scenarios do not require deep nesting. However, there is one exception: list-types in multi-modal documents look like a single nesting to the user, but are actually two levels of nesting under the hood. This means that there is no straightforward way to get the root doc from a list doc.
> 
> So the new plan is the following:
> 
> * We make `root_id` as multimodal-only feature
> * This will be a `@property` so we don't have to change the proto. Under the hood it will access a field in `_metadata`
> * The information in `_metadata` is set only once, when the dataclass is converted to Document. We make no guarantees about any behaviour if the user changes the document tree structure afterwards

I do not like this, this will make this feature almost not accessible from non-python clients. Changing the proto should not be a big problem if the root_id is passed at the end of the list of attributes.

I would say, either we go with it or we do not, we do not go only half way

**Comment by JohannesMessner** [comment]
Ok I see your point; the main reason behind making it a property was because in this design it would be multi-modal only. If it is a real field but not really supported for vanilla Documents that would be weird imo.

Let me think about a way that we can "go for it" with multi-modal and vanilla docs, while also avoiding the big overheads. Otherwise I'll put it on hold again.

**Comment by JoanFM** [comment]
Will not do as it is
